entities/src/main/java/com/flipkart/varadhi/entities/GetMessagesRequest.java

```diff
@@ -0,0 +1,35 @@
+package com.flipkart.varadhi.entities;
+
+
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+public class GetMessagesRequest {
+    public static final long UNSPECIFIED_TS = 0L;
+    public static final long LATEST_TS = Long.MAX_VALUE;
+
+    //TODO:: evaluate if earliestFailedAt should be per shard level to support pagination kind of semantics.
```

Comment:
i am hoping that shard semantics, is totally transparent and is not leaking into public api. Shard is just "a" implementation of scaling subscription, but not the only one. sharding strategies may also differ. so an operation suitable for one shard strategy might not work for another shard strategy.

So, I was hoping that most operations are agnostic of "shards", if any operation needs to be mapped then it is done either by "server" or "controller". most likely controller as that guy has the know how.